appdaemontestframework/init_framework.py

```diff
@@ -47,7 +47,11 @@ def patch_hass():
     patches = []
     hass_functions = {}
     for function_name in actionable_functions_to_patch:
-        patch_function = mock.patch.object(Hass, function_name, create=True)
+        autospec = False
+        # spec the __init__ call se we get access to the instance object in the mock
+        if function_name == '__init__':
+            autospec = True
```

Comment:
Maybe add a function `should_autospect(function_name)`, so that we encapsulate the logic there and it doesn't clutter the main patching logic. Then  in the function we can explain why `__init__` should be autospeced, just like you did 🙂

```python
def should_autospec(function_name):
  # spec the __init__ call se we get access to the instance object in the mock 
  return function_name == '__init__'
```

And then

```python
patch_function = mock.patch.object(Hass, function_name, create=True, autospec=should_autospec(function_name))
```

Reply:
Instead of creating an individual list for each thing we may need to do, what if there was one list that holds objects that have all info for any object to be mocked. I think I'm going to need something like this in my work on #18 anyway. I'll take a stab at this and push it up so you can see what I mean. Let me know what you think. (I won't be able to get to this until later today)

Reply:
Fantastic! I love your approach 👍

Please, just remove the 2 leftover `print` statements on line 58-59 and I'm merging 🙂

Reply:
all set